client/elements/suttaplex/sc-suttaplex.html

```diff
@@ -577,7 +584,7 @@ <h2 class="not-found-error">{{localize('itemNotFound')}}</h2>
                   });
                   return scAcronym.replace(/–\s*$/, '');
               } catch (e) {
-                  return;
+                  console.error(e);
```

Comment:
Why do you do a try-catch here and not in the same function in sc-parallel-list.html

Reply:
the try/catch was already here, I just printed the error. By the way, if the functions are exactly identical, we should definitely consider moving them to a separate element that would have utility functions to follow [https://en.wikipedia.org/wiki/Don%27t_repeat_yourself](DRY).

This `_transformId` function is an ideal candidate to move it to a separate utility class, because it's a pure function. Pure functions only depend on their arguments and don't have side effects. 
Example of a pure function: 
```
add2(x) {
   return 2 + x;
}
```
`add2` is a pure function, because when called with `x=1`, it will always return `3`, Similarly, `_transformId(rootId, expansionData)` will always return the same value for the same arguments and it doesn't use any external variables in the function body.

Pure functions are easy to maintain and don't break stuff when called.

Reply:
The functions are not entirely identical. There are a few small differences. But I guess nothing that cannot be overcome by adding a few attributes.